backend/typescript/services/implementations/camperService.ts

```diff
@@ -0,0 +1,69 @@
+import ICamperService from "../interfaces/camperService";
+import MgCamper, { Camper } from "../../models/camper.model";
+import { CamperDTO, UpdateCamperDTO } from "../../types";
+import { getErrorMessage } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+
+const Logger = logger(__filename);
+
+class CamperService implements ICamperService {
+  /* eslint-disable class-methods-use-this */
+  async updateCamperById(
+    camperId: string,
+    camper: UpdateCamperDTO,
+  ): Promise<CamperDTO> {
+    let oldCamper: Camper | null;
+
+    try {
+      // must explicitly specify runValidators when updating through findByIdAndUpdate
+      oldCamper = await MgCamper.findByIdAndUpdate(
```

Comment:
nit but can we rename oldCamper => updatedCamper since this object actually contains the updated fields?